dataherald/services/sql_generations.py

```diff
@@ -75,3 +81,17 @@ def create(
 
     def get(self, query) -> list[SQLGeneration]:
         return self.sql_generation_repository.find_by(query)
+
+    def execute(self, sql_generation_id: str, query: Query) -> tuple[str, dict]:
+        sql_generation_repository = SQLGenerationRepository(self.storage)
```

Comment:
you don't need to intatiated again the repository it was initialized in the __init__
just call `self.sql_generation_repository`

Reply:
fixed